examples/pubsub.rs

```diff
@@ -57,39 +59,39 @@ impl subscriber::FutureService for Subscriber {
 }
 
 impl Subscriber {
-    fn listen(id: u32) -> SocketAddr {
+    fn listen(id: u32, options: server::Options) -> SocketAddr {
         Subscriber { id: id }
-            .listen("localhost:0".first_socket_addr(),
-                    server::Options::default())
-            .wait()
+            .listen("localhost:0".first_socket_addr(), options)
             .unwrap()
     }
 }
 
 #[derive(Clone, Debug)]
 struct Publisher {
-    clients: Arc<Mutex<HashMap<u32, subscriber::FutureClient>>>,
+    clients: Rc<RefCell<HashMap<u32, subscriber::FutureClient>>>,
```

Comment:
Sure. It's fine to leave it with no comment. My thought process is like this: `RefCell` only performs runtime checks on mutable borrows, and therefore afaiu you can't quite declare victory once this compiles.